### DISCUSSION TITLE
why is the tidy.table class needed?

### DISCUSSION THREAD

**Discussion by bnicenboim** [markfairbanks/tidytable#314] [Q&A] [answered]
Hi,
so I was wondering why is the new class required? Is there something special about tidy.tables that is different from regular data.table?

**Comment by markfairbanks** [comment, marked as answer]
The tidytable class was created so that printing would match tidyverse tibbles

<img width="310" alt="image" src="https://user-images.githubusercontent.com/28766000/139443967-377c3239-ab2b-40d5-b738-8e6ed1f3c85e.png">

**Comment by bnicenboim** [comment]
ahh, ok.

It would be nice that if I use `mutate.` on a data.table that it would return a data.table as well, rather than change the class of the object.
This is specially important in what I'm doing because my objects are of the class `c("signal_tbl", "data.table", "data.frame")`, and when I use `mutate.` I get as an output `c("tidytable" ,"data.table", "data.frame")` rather than my original classes.

**Comment by markfairbanks** [comment]
This should be solved by making your class a subclass of a tidytable `c("signal_tbl", "tidytable", "data.table", "data.frame")`.

After that you'll be able to use `mutate.()` with no issues.

``` r
library(tidytable)

df <- data.table(x = 1:3, y = 1:3)

class(df) <- c("signal_tbl", "tidytable", "data.table", "data.frame")

df %>%
  mutate.(double_x = x * 2) %>%
  class()
#> [1] "signal_tbl" "tidytable"  "data.table" "data.frame"
```

**Comment by markfairbanks** [comment]
Actually I forgot to ask - is this your own custom class? Or does it belong to another package?

**Comment by bnicenboim** [comment]
Yes my own in bnicenboim/eeguana

I'm trying to clean up my messy code and just use your functions now, and
I'm trying the unit tests right now.

Would you be against  .data.table versions of your functions that just
return a data table and don't add the tidy table class?

**Comment by markfairbanks** [comment]
> Would you be against  .data.table versions of your functions that just return a data table and don't add the tidy table class?

Unfortunately at this point I'm wary to make a change like this. tidytable functions have converted data.tables to tidytables for almost 2 years. I think it would be a pretty unexpected change for users to have them suddenly return data.tables.

Since it is your own package, I would recommend adjusting your class to be a subclass of tidytable.

FYI you'll still be able to use data.table syntax on them if you want:

``` r
library(tidytable)

df <- tidytable(x= 1:3, y = c("a", "a", "b"))

df[, .(avg_x = mean(x)), by = y]
#> # A tidytable: 2 × 2
#>   y     avg_x
#>   <chr> <dbl>
#> 1 a       1.5
#> 2 b       3
```

**Comment by bnicenboim** [comment]
I see. No problem!
